homeassistant/components/binary_sensor/wirelesstag.py

```diff
@@ -0,0 +1,208 @@
+
+# On means open, Off means closed
+SENSOR_DOOR = 'door'
+
+# On means temperature become too cold, Off means normal
+SENSOR_COLD = 'cold'
+
+# On means hot, Off means normal
+SENSOR_HEAT = 'heat'
+
+# On means too dry (humidity), Off means normal
+SENSOR_DRY = 'dry'
+
+# On means too wet (humidity), Off means normal
+SENSOR_WET = 'wet'
+
+# On means light detected, Off means no light
+SENSOR_LIGHT = 'light'
+
+# On means moisture detected (wet), Off means no moisture (dry)
+SENSOR_MOISTURE = 'moisture'
+
+# On means tag battery is low, Off means normal
+SENSOR_BATTERY = 'low_battery'
+
+# Sensor types: Name, device_class, push notification type representing 'on',
+# attr to check
+SENSOR_TYPES = {
+    SENSOR_PRESENCE: ['Presence', 'presence', 'is_in_range', {
+        "on": "oor",
+        "off": "back_in_range"
+        }, 2],
+    SENSOR_MOTION: ['Motion', 'motion', 'is_moved', {
+        "on": "motion_detected",
+        }, 5],
+    SENSOR_DOOR: ['Door', 'door', 'is_door_open', {
+        "on": "door_opened",
+        "off": "door_closed"
+        }, 5],
+    SENSOR_COLD: ['Cold', 'cold', 'is_cold', {
+        "on": "temp_toolow",
+        "off": "temp_normal"
+        }, 4],
+    SENSOR_HEAT: ['Heat', 'heat', 'is_heat', {
+        "on": "temp_toohigh",
+        "off": "temp_normal"
+        }, 4],
+    SENSOR_DRY: ['Too dry', 'dry', 'is_too_dry', {
+        "on": "too_dry",
+        "off": "cap_normal"
+        }, 2],
+    SENSOR_WET: ['Too wet', 'wet', 'is_too_humid', {
+        "on": "too_humid",
+        "off": "cap_normal"
+        }, 2],
+    SENSOR_LIGHT: ['Light', 'light', 'is_light_on', {
+        "on": "too_bright",
+        "off": "light_normal"
+        }, 1],
+    SENSOR_MOISTURE: ['Leak', 'moisture', 'is_leaking', {
+        "on": "water_detected",
+        "off": "water_dried",
+        }, 1],
+    SENSOR_BATTERY: ['Low Battery', 'battery', 'is_battery_low', {
+        "on": "low_battery"
+        }, 3]
+}
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ENTITY_NAMESPACE, default=DEFAULT_ENTITY_NAMESPACE):
+        cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the platform for a WirelessTags."""
+    platform = hass.data.get(WIRELESSTAG_DOMAIN)
+
+    sensors = []
+    tags = platform.tags
+    for _, tag in tags.items():
+        allowed_sensor_types = WirelessTagBinarySensor.allowed_sensors(tag)
+        for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+            if sensor_type in allowed_sensor_types:
+                sensors.append(WirelessTagBinarySensor(platform, tag,
+                                                       sensor_type))
+
+    platform.install_push_notifications(sensors)
+    add_devices(sensors, True)
+
+
+class WirelessTagBinarySensor(WirelessTagBaseSensor, BinarySensorDevice):
```

Comment:
You probably want to implement the `device_class` property so that they get the right icons in the frontend.

Reply:
got it. in progress

Reply:
done.